anndata/utils.py

```diff
@@ -254,3 +267,18 @@ def func(*_, **__):
             raise error
 
     return func
+
+
+@singledispatch
+def dim_len(x, dim):
+    return x.shape[dim]
+
+
+@dim_len.register(ak.Array)
+def dim_len_array(x, dim):
+    if dim == 0:
+        return len(x)
+    elif dim == 1:
+        return None
+    else:
+        raise IndexError()
```

Comment:
Consider generalizing that this returns a value for all fixed-length dimensions and None for variable-length dimensions.